lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -2256,6 +2256,77 @@ struct FoldResetMux : public mlir::RewritePattern {
 };
 } // namespace
 
+namespace {
+/// This canonicalizer provides the following patterns:
+///   reset(reg) = 0 ==> connect(reg, and(reg, x)) ==> reg -> 0
+///   reset(reg) = 0 ==> connect(reg, and(x, reg)) ==> reg -> 0
+///   reset(reg) = 1 ==> connect(reg, or(reg, x))  ==> reg -> 1
+///   reset(reg) = 1 ==> connect(reg, or(x, reg))  ==> reg -> 1
```

Comment:
Mega-nit: This syntax is bothering me and took me a while to figure it out. The `==>` is an "AND" in the first use and is "is converted to" (or "IMPLIES"?) in the second.

Reply:
Ah OK I can delete these comments, I don't think they're helpful. FWIW `==>` is implication, `->` is canonicalization.